servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/RetryingHttpRequesterFilter.java

```diff
@@ -745,6 +795,11 @@ public static final class Builder {
         @Nullable
         private RetryCallbacks onRequestRetry;
 
+        public Builder returnFailedResponses(final boolean returnFailedResponses) {
+            this.returnFailedResponses = returnFailedResponses;
+            return this;
+        }
```

Comment:
I'm certain this can have a better name and clearly it needs docs before merging. Name suggestions welcome.

Reply:
I also think this API is a bit awkward: first you must turn a response into an HttpResponseException and then it's going to be discarded. Alternatively, we could just have a different lambda to the tune of `Function<Boolean, HttpResponseMetadata> shouldRetry`.

Reply:
Right now we don't have RS operators to achieve retries without mapping into exceptions. If we go the route of clean retry of response meta-data without mapping to exceptions, it's possible but will take longer. 

Current rational was that some users want to always map responses to exceptions, that's why we have independent `responseMapper`. Then some users may want to retry that, so there is a 2nd method for them to `retryResponses`. We decided to put them next to each other on the same builder instead of offering 2 different filters bcz they often used together.

I agree that having a 3rd method that works only if the other 2 also configured is not intuitive. Alternatively, we can consider adding a `retryResponses` overload that takes a `boolean` to make a decision if it need to unwrap the original response or not.

Reply:
I like the idea of the boolean overload, which would signal that it needs to be configured "together". Alternatively when building, we should at least check if this value is set to true and others are in their default state to reject the config?